python/pyspark/context.py

```diff
@@ -306,6 +306,14 @@ def _do_init(
         self.pythonExec = os.environ.get("PYSPARK_PYTHON", "python3")
         self.pythonVer = "%d.%d" % sys.version_info[:2]
 
+        if sys.version_info[:2] < (3, 8):
```

Comment:
just curious whether the version number could be something different from 3.7? if so perhaps the warning message  could be something like `Python support for version <= 3.7 is deprecated in Spark 3.4.`

Reply:
Same though at first glance. But as we only support Python3.7+, I think there might be other version check so here it is 3.7 for sure? Not sure. If not, then revised error message would be better.

Reply:
Good question. It's unlikely because we did this since Apache Spark 3.2 for a while. This is the first step and we did many efforts to kill 3.6 completely. :)

https://github.com/apache/spark/blob/0d11085e682eefb19cb4e6882b217a1225c27dbc/python/pyspark/context.py#L233-L239

Reply:
In addition, Spark 3.4 requires Python 3.7+ at the installation stage explicitly.
https://github.com/apache/spark/blob/8628c15d176a8084b695bbb6ca841831e6fb75d4/python/setup.py#L277

Reply:
I see, never mind then :)